File: cpp/program/Printer.cpp
#pragma once
#include <iostream>
#include <string>
#include "Peripheral.cpp"

using namespace std;

class Printer : public Peripheral{
private:
    string jenisPrinter;
    int kecepatanCetak;
public:
    Printer(){};
    
    Printer(string jenisPrinter, int kecepatanCetak, string merk, string jenis) : Peripheral(merk, jenis){
        this->jenisPrinter = jenisPrinter;
        this->kecepatanCetak = kecepatanCetak;
    }
    
    void setjenisPrinter(string jenisPrinter){
        this->jenisPrinter = jenisPrinter;
    }
    void setkecepatanCetak(int kecepatanCetak){
        this->kecepatanCetak = kecepatanCetak;
    }

    string getjenisPrinter(){
        return this->jenisPrinter;
    }
    int getkecepatanCetak(){
        return this->kecepatanCetak;
    }
    
    ~Printer(){};
};

File: python/program/Printer.py
from Peripheral import Peripheral

class Printer(Peripheral):
    def __init__(self, jenis_printer='', kecepatan_cetak=0, merk='', jenis=''):
        super().__init__(merk, jenis)
        self.jenis_printer = jenis_printer
        self.kecepatan_cetak = kecepatan_cetak

    def set_jenis_printer(self, jenis_printer):
        self.jenis_printer = jenis_printer

    def set_kecepatan_cetak(self, kecepatan_cetak):
        self.kecepatan_cetak = kecepatan_cetak

    def get_jenis_printer(self):
        return self.jenis_printer

    def get_kecepatan_cetak(self):
        return self.kecepatan_cetak
